### ISSUE TITLE
Fatal error & uncaught error with WooCommerce Product Bundles plugin

### ISSUE THREAD

**Issue by Orhn** [pojome/elementor#5338]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description

Fatal error, uncaught error, when using WooCommerce Product Bundles plugin. It was working with RC4.

## Steps to reproduce

Enable plugin.

## Isolating the problem

- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Environment

<details>
<summary>System Info</summary>
					
== Server Environment ==
	Operating System: Linux
	Software: nginx/1.10.1
	MySQL version: 5.6.37
	PHP Version: 7.1.7
	PHP Max Input Vars: 4000
	PHP Max Post Size: 1000M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 4.9.8
	Site URL: https://ecommerce-v2.local
	Home URL: https://ecommerce-v2.local
	WP Multisite: No
	Max Upload Size: 300 MB
	Memory limit: 40M
	Permalink Structure: /%postname%/
	Language: en-US
	Timezone: Europe/Stockholm
	Debug Mode: Inactive

== Theme ==
	Name: Elementor Hello Theme
	Version: 1.0
	Author: Elementor Team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/11.1.2 Safari/605.1.15

== Active Plugins ==
	Elementor
		Version: 2.1.6
		Author: Elementor.com

	Elementor Pro
		Version: 2.1.0
		Author: Elementor.com

	WooCommerce
		Version: 3.4.4
		Author: Automattic

	WooCommerce Product Add-ons
		Version: 2.9.6
		Author: WooCommerce

	WooCommerce Product Bundles
		Version: 5.7.11
		Author: SomewhereWarm

== Must-Use Plugins ==
	Local by Flywheel Relative URL (for Live Links)
		Version: 1.0
		Author: Flywheel

== Debug ==
	Errors: 
		TypeError: null is not an object (evaluating 'this.dataAdapter.query')
			at https://ecommerce-v2.local/wp-content/plugins/elementor/assets/lib/e-select2/js/e-select2.full.js?ver=4.0.6-rc.1 - 5449:23
			2018-08-08 08:02 +00:00
			x 2 times

		TypeError: self.attributes[ control.name ].each is not a function. (In 'self.attributes[ control.name ].each( function( item ) {
					styleFields.push( self.getStyleControls( control.fields, item.attributes ) );
				} )', 'self.attributes[ control.name ].each' is undefined)
			at https://ecommerce-v2.local/wp-content/plugins/elementor/assets/js/editor.js?ver=2.1.6 - 6414:41
			2018-08-06 07:15 +00:00
			x 4 times

		TypeError: null is not an object (evaluating 'this.dataAdapter.query')
			at https://ecommerce-v2.local/wp-content/plugins/elementor/assets/lib/e-select2/js/e-select2.full.min.js?ver=4.0.6-rc.1 - 1:60823
			2018-08-02 08:27 +00:00
			x 2 times

		TypeError: null is not an object (evaluating 'e.editor.on')
			at https://ecommerce-v2.local/wp-content/plugins/elementor/assets/js/editor.min.js?ver=2.1.6 - 2:82960
			2018-08-01 07:50 +00:00

		TypeError: null is not an object (evaluating 'this.dataAdapter.query')
			at https://ecommerce-v2.local/wp-content/plugins/elementor/assets/lib/e-select2/js/e-select2.full.min.js?ver=4.0.6-rc.1 - 1:60823
			2018-08-01 07:41 +00:00

		TypeError: undefined is not an object (evaluating 'this.ui.select.select2')
			at https://ecommerce-v2.local/wp-content/plugins/elementor/assets/js/editor.min.js?ver=2.1.6 - 2:76869
			2018-07-31 14:05 +00:00
			x 2 times

		TypeError: null is not an object (evaluating 'this.dataAdapter.query')
			at https://ecommerce-v2.local/wp-content/plugins/elementor/assets/lib/e-select2/js/e-select2.full.min.js?ver=4.0.6-rc.1 - 1:60823
			2018-07-30 10:16 +00:00

		TypeError: o.attributes[t.name].each is not a function. (In 'o.attributes[t.name].each(function(e){n.push(o.getStyleControls(t.fields,e.attributes))})', 'o.attributes[t.name].each' is undefined)
			at https://ecommerce-v2.local/wp-content/plugins/elementor/assets/js/editor.min.js?ver=2.1.4 - 2:105707
			2018-07-26 04:02 +00:00
			x 4 times

		TypeError: undefined is not an object (evaluating 'this.ui.select.select2')
			at https://ecommerce-v2.local/wp-content/plugins/elementor/assets/js/editor.min.js?ver=2.1.4 - 2:76862
			2018-07-25 13:35 +00:00
			x 4 times

		TypeError: undefined is not an object (evaluating 'i.attributes')
			at https://ecommerce-v2.local/wp-content/plugins/elementor/assets/js/frontend.min.js?ver=2.1.4 - 2:7459
			2018-07-25 10:55 +00:00
			x 2 times

				
</details>

---

## Error

<details>
<summary>Error output</summary>
<p>Fatal error: Uncaught Error: Call to a member function is_type() on string in /app/public/wp-content/plugins/woocommerce-product-bundles/includes/class-wc-pb-display.php:294 Stack trace: #0 /app/public/wp-includes/class-wp-hook.php(286): WC_PB_Display->add_edit_in_cart_notice('') #1 /app/public/wp-includes/class-wp-hook.php(310): WP_Hook->apply_filters('', Array) #2 /app/public/wp-includes/plugin.php(453): WP_Hook->do_action(Array) #3 /app/public/wp-content/plugins/elementor-pro/modules/woocommerce/documents/product.php(133): do_action('woocommerce_bef...') #4 /app/public/wp-content/plugins/elementor-pro/modules/theme-builder/classes/locations-manager.php(216): ElementorPro\Modules\Woocommerce\Documents\Product->print_content() #5 /app/public/wp-content/plugins/elementor-pro/modules/theme-builder/classes/locations-manager.php(169): ElementorPro\Modules\ThemeBuilder\Classes\Locations_Manager->do_location('single') #6 [internal function]: ElementorPro\Modules\ThemeBuilder\Classes\Locations_Manager->ElementorPro\Modules\Them in /app/public/wp-content/plugins/woocommerce-product-bundles/includes/class-wc-pb-display.php on line 294</p>
</details>

**Comment by teodorboev** [comment]
Same here, I took me all morning debugging to figure out what was the culprit... I can confirm that it doesn't matter what theme you have active, if you try to do a WC Single Product, Elementor will not load...

**Comment by arielk** [comment]
Is this plugin in the WordPress repository?
I'd appreciate it if you send us a link to the plugin that we can check it.

**Comment by Orhn** [comment]
https://woocommerce.com/products/product-bundles/#

**Comment by bainternet** [comment]
@Orhn  @teodorboev 

can you try adding 
```PHP
global $product;
```

to the top of `print_content` method in `elementor-pro/modules/woocommerce/documents/product.php` line 132

```PHP
public function print_content() {
	do_action( 'woocommerce_before_single_product' );
	parent::print_content();
	do_action( 'woocommerce_after_single_product' );
}
```

so it becomes:
```PHP
public function print_content() {
	global $product;
	do_action( 'woocommerce_before_single_product' );
	parent::print_content();
	do_action( 'woocommerce_after_single_product' );
}
```

and let us know.

**Comment by Orhn** [comment]
@bainternet that didn't fix.

---

Fatal error: Uncaught Error: Call to a member function is_type() on string in /app/public/wp-content/plugins/woocommerce-product-bundles/includes/class-wc-pb-display.php on line 294

Error: Call to a member function is_type() on string in /app/public/wp-content/plugins/woocommerce-product-bundles/includes/class-wc-pb-display.php on line 294

**Comment by bainternet** [comment]
[**@Orhn**](https://github.com/Orhn) commented on [Sep 3, 2018, 10:04 AM GMT+3](https://github.com/pojome/elementor/issues/5338#issuecomment-418020205 "2018-09-03T07:04:22Z - Replied by Github Reply Comments"):
> @bainternet that didn't fix.
> 
> ---
> 
> Fatal error: Uncaught Error: Call to a member function is_type() on string in /app/public/wp-content/plugins/woocommerce-product-bundles/includes/class-wc-pb-display.php on line 294
> 
> Error: Call to a member function is_type() on string in /app/public/wp-content/plugins/woocommerce-product-bundles/includes/class-wc-pb-display.php on line 294

thanks for helping out, one last try can you try this:

```PHP
public function print_content() {
	global $product;
	if ( ! is_object( $product)) $product = wc_get_product( get_the_ID() );

	do_action( 'woocommerce_before_single_product' );
	parent::print_content();
	do_action( 'woocommerce_after_single_product' );
}
```

**Comment by Orhn** [comment]
@bainternet 

Yes it works! Thanks a lot.

**Comment by Orhn** [comment]
@bainternet 

Another issue just popped up. When editing Single Product Template this message shows up:

The preview could not be loaded. We're sorry, but something went wrong. Click on 'Learn more' and follow each of the steps to quickly solve it.

Disabling the plugin solves the issue.

**Comment by arielk** [comment]
Merged into Pro v2.1.8, please update.

Thanks for your feedback!
